### ISSUE TITLE
Can't specify all options when using Kubernetes clusters withVersion

### ISSUE THREAD

**Issue by umbrant** [Azure/azure-sdk-for-java#1989]
I'm modifying the example found here:

https://github.com/Azure-Samples/aks-java-manage-kubernetes-cluster/blob/master/src/main/java/com/microsoft/azure/management/kubernetescluster/samples/ManageKubernetesCluster.java

```java
            KubernetesCluster kubernetesCluster = azure.kubernetesClusters().define(aksName)
                .withRegion(region)
                .withNewResourceGroup(rgName)
                .withLatestVersion()
                .withRootUsername(rootUserName)
                .withSshKey(sshKeys.getSshPublicKey())
                .withServicePrincipalClientId(servicePrincipalClientId)
                .withServicePrincipalSecret(servicePrincipalSecret)
                .defineAgentPool("agentpool")
                    .withVirtualMachineCount(1)
                    .withVirtualMachineSize(ContainerServiceVMSizeTypes.STANDARD_D1_V2)
                    .attach()
                .withDnsPrefix("dns-" + aksName)
                .create();
```

If you change withLatestVersion to withVersion, the return type changes to a withDnsPrefix which then goes to a withCreate, meaning we're left with:

```java
            KubernetesCluster kubernetesCluster = azure.kubernetesClusters().define(aksName)
                .withRegion(region)
                .withNewResourceGroup(rgName)
                .withVersion(KubernetesVersion.KUBERNETES_1_8_1)
                .withDnsPrefix("dns-" + aksName)
                .create();
```

Is this intentional? I'd still like to specify all the options.

**Comment by milismsft** [comment]
nope, that is a bug. I will address it in the next release.

**Comment by milismsft** [comment]
I have a fix ready for this (see https://github.com/Azure/azure-libraries-for-java/pull/79)

@umbrant Let me know if this bug is blocking you.

**Comment by umbrant** [comment]
Thanks for the quick fix @milismsft ! I'm not immediately blocked since I can use withLatestVersion for now, though it'd be great to see this in a released version.

**Comment by milismsft** [comment]
BTW it's safer in the long term to use .withLatestVersion() since the service is in preview and it can be updated at any time and it can be a while for the SDK to catch up with those changes.

**Comment by milismsft** [comment]
Fixed in v1.5
